src/store/reducers/platform.js

```diff
@@ -41,35 +43,41 @@ const initialState = fromJS({
   }
 });
 
-const Init = createAction("PLATFORM/INIT", async () => null);
-
+const Init = createAction("PLATFORM/INIT");
 const platformInitEpic = () => async dispatch => {
   dispatch(Init());
 };
 
 const setPlatformErrors = createAction("PLATFORM/SET_PLATFORM_ERRORS");
-
 const resetPlatformErrors = createAction("PLATFORM/RESET_PLATFORM_ERRORS");
 
-const web3Initialized = createAction("PLATFORM/WEB3_INITIALIZED", p => p);
-
-const web3Reset = createAction("PLATFORM/WEB3_RESET", () => web3.reset());
+const setActiveNodeType = createAction('PLATFORM/SET_ACTIVE_PROVIDER_TYPE');
 
-const web3ResetKeepSync = createAction("PLATFORM/WEB3_RESET_KEEP_SYNC", () =>
-  web3.reset(true)
-);
+const web3Initialized = createAction("PLATFORM/WEB3_INITIALIZED");
+const web3Reset = createAction("PLATFORM/WEB3_RESET", () => {
+  try {
+    web3.reset(false);
+  } catch (e) {
+    Promise.promisify(web3.reset).call(false);
```

Comment:
Not quite sure how this solves the problem?
You will still invoke web3.reset which is implemented as sync call.

Reply:
If the first call fails then we do second one which uses callback.